ledger/complete/wal/checkpoint_v6_reader.go

```diff
@@ -520,3 +536,14 @@ func getNodeByIndex(subtrieNodes [][]*node.Node, topLevelNodes []*node.Node, ind
 	// +1 because first item is always nil
 	return topLevelNodes[offset+1], nil
 }
+
+// reachedEOF checks if the reader has reached end of file
+// it returns nil if reached EOF
+// any error returned are exception
+func reachedEOF(reader io.Reader) error {
+	_, err := reader.Read(make([]byte, 1))
```

Comment:
i'd rather using `File.Stat()` to use file size check EOF. so if in future we need to call `read` after `reachedEOF`, we won't miss the byte read here.

Reply:
you can't use File here since it's a io.Reader, which may or may not be a file.

Changing the argument to take an io.ReadSeeker instead of io.Reader would enable to seek position to determine EOF